nginx-auth0/nginx.conf

```diff
@@ -155,12 +160,24 @@ http {
          if res.email then
            ngx.log(ngx.DEBUG, "bearer_jwt_verify response: mail ", res.email)
            ngx.req.set_header("X-Akvo-Email", res.email)
+           ngx.var.akvoemail = res.email
          else
            ngx.say(cjson.encode({error="Invalid access_token"}))
            ngx.exit(ngx.HTTP_FORBIDDEN)
 	 end
       }
 
+      proxy_cache http_204;
+      proxy_cache_revalidate off;
+      proxy_cache_background_update on;
+      proxy_cache_lock on;
+      proxy_cache_methods GET;
+      proxy_cache_key $proxy_host$request_uri$akvoemail;
```

Comment:
Using `$request_uri` means that the user can use any cache busting param to bypass the cache. 

Is it possible to pick up just the http params that we support? This will cause some coupling between the backend code and the nginx one, so maybe I am just being paranoid and it is not really needed.

Reply:
I don't think we need to invest way too much on hardening the key based on _supported parameters_. We wanted something easier to implement than _Long poll_.

My intention is not to document that there is caching going on. The contract will be that after getting HTTP 204, the client must stop requesting and _wait_ on its side. This caching will be just a safety net for those clients that will not wait on their side.